torch/csrc/jit/mobile/backport_v5_to_v4.cpp

```diff
@@ -0,0 +1,131 @@
+#include <ATen/core/ivalue.h>
+//#include <caffe2/serialize/file_adapter.h>
+#include <caffe2/serialize/inline_container.h>
+#include <torch/csrc/jit/mobile/backport_factory.h>
+#include <torch/csrc/jit/mobile/model_compatibility.h>
+#include <torch/csrc/jit/mobile/module.h>
+#include <torch/csrc/jit/serialization/pickler.h>
+
+namespace torch {
+namespace jit {
+
+using c10::IValue;
+using caffe2::serialize::PyTorchStreamReader;
+using caffe2::serialize::PyTorchStreamWriter;
+using caffe2::serialize::ReadAdapterInterface;
+using torch::jit::TensorIndexMap;
+
+namespace {
+TensorIndexMap get_tensors_archive_table(const IValue& value) {
+  std::vector<char> data;
+  TensorIndexMap tensors_archive_table;
+  // Vector to capture the run-time class types during pickling the IValues
+  std::vector<c10::ClassTypePtr> memoizedClassTypes;
+  Pickler data_pickle(
+      [&](const char* buf, size_t size) {
+        data.insert(data.end(), buf, buf + size);
+      },
+      nullptr,
+      nullptr,
+      &memoizedClassTypes);
+  data_pickle.protocol();
+  data_pickle.pushIValue(value);
+  data_pickle.stop();
+
+  const auto tensor_candidates = data_pickle.tensorData();
+  for (size_t tensor_index = 0; tensor_index < tensor_candidates.size();
+       tensor_index++) {
+    tensors_archive_table[tensor_candidates[tensor_index]] =
+        std::make_pair(kArchiveNameConstants, tensor_index);
+  }
+  return tensors_archive_table;
+}
+
+void writeArchive(
+    PyTorchStreamWriter& writer,
+    const std::string& archive_name,
+    const IValue& value,
+    const TensorIndexMap& tensors_archive_table,
+    bool use_tensors_archive_table) {
```

Comment:
IF this is v5 to v4 specific do you need `use_tensors_archive_table` arg?

Reply:
Good catch! No I don't need it. Also remove the unused logic in the code.